src/Generators/randomgenerators.jl

```diff
@@ -0,0 +1,407 @@
+
+### Optional Parameters
+- `A::Real`: probability of edge in partition A (default `0.57`)
+- `B::Real`: probability of edge in partition B (default `0.19`)
+- `C::Real`: probability of edge in partition C (default `0.19`)
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+
+###
+References
+- http://www.graph500.org/specifications#alg:generator
+"""
+struct Kronecker{T<:Integer, U<:Real, R<:AbstractRNG} <: RandomGenerator
+    SCALE::T
+    edgefactor::T
+    A::U
+    B::U
+    C::U
+    rng::R
+end
+
+Kronecker(SCALE, edgefactor; A=0.57, B=0.19, C=0.19, rng=GLOBAL_RNG) = Kronecker(SCALE, edgefactor, A, B, C, rng)
+
+"""
+    struct DorogovtsevMendes <: RandomGenerator dorogovtsev_mendes(n)
+
+A struct representing a generator for random `n` vertex graph using the Dorogovtsev-Mendes method (with `n \\ge 3`).
+
+The Dorogovtsev-Mendes process begins with a triangle graph and inserts `n-3` additional vertices.
+Each time a vertex is added, a random edge is selected and the new vertex is connected to the two
+endpoints of the chosen edge. This creates graphs with a many triangles and a high local clustering coefficient.
+
+It is often useful to track the evolution of the graph as vertices are added, you can access the graph from
+the `t`th stage of this algorithm by accessing the first `t` vertices with `g[1:t]`.
+
+### Required Fields
+- `n::Integer`: The number of vertices in the graph
+
+### Optional Parameters
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+
+### References
+- http://graphstream-project.org/doc/Generators/Dorogovtsev-Mendes-generator/
+- https://arxiv.org/pdf/cond-mat/0106144.pdf#page=24
+"""
+struct DorogovtsevMendes{T<:Integer, R<:AbstractRNG} <: RandomGenerator
+    n::T
+    rng::R
+    function DorogovtsevMendes(n::T, rng::R) where {T<:Integer, R<:AbstractRNG}
+        n < 3 && throw(DomainError("n=$n must be at least 3"))
+        new{T, R}(n, rng)
+    end
+end
+DorogovtsevMendes(n; rng=GLOBAL_RNG) = DorogovtsevMendes(n, rng)
+
+"""
+    struct RandomOrientationDAG <: RandomGenerator
+
+Generate a random oriented acyclical digraph. The function takes in a simple
+graph and an optional random number generator as an argument. The probability
+of each directional acyclic graph randomly being generated depends on the
+architecture of the original directed graph.
+
+### Required Fields
+- `g::AbstractGraph`: the original graph
+
+### Optional Parameters
+- `rng::AbstractRNG`: set the random number generator (default: `Random.GLOBAL_RNG`)
+
+### Implementation Notes
+- The graph `g` does not have to be of the same type as the output.
+- DAG's have a finite topological order; this order is randomly generated via "order = randperm()".
+"""
+struct RandomOrientationDAG{T<:Integer, AG<:AbstractGraph{T}, R<:AbstractRNG} <: RandomGenerator
+    g::AG
+    rng::R
+end
+RandomOrientationDAG(g; rng=GLOBAL_RNG) = RandomOrientationDAG(g, rng)
```

Comment:
This struct is a bit weird, because it is more of an operator than a random graph.

Reply:
Agreed. Let's flag this and discuss moving it out in a separate PR. I think it makes sense to move it.